src/client.ts

```diff
@@ -786,45 +748,39 @@ export class SpvWalletClient {
    * @return {string} Final transaction hex
    */
   FinalizeTransaction(draftTransaction: DraftTransaction): string {
-    if (!this.options?.xPriv) {
-      const Err = new Error('cannot sign transaction without an xPriv')
-      this.logger.error(Err.message)
-      throw Err
+    if (!this?.xPriv) {
+      throw new ErrorNoXPrivToSignTransaction();
     }
 
-    const Input = bsv.Transaction.Input
-    const xPriv = this.options.xPriv as bsv.HDPrivateKey
-    const txDraft = new bsv.Transaction(draftTransaction.hex)
+    const Input = bsv.Transaction.Input;
```

Comment:
### minor
```suggestion
    const input = bsv.Transaction.Input;
```

Reply:
I've removed this line and used the `bsv.Transaction.Input` inline